SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -185,6 +187,41 @@ void USpatialNetDriver::OnMapLoaded(UWorld* LoadedWorld)
 	Connect();
 }
 
+void USpatialNetDriver::OnLevelAddedToWorld(ULevel* LoadedLevel, UWorld* OwningWorld)
+{
+	// If we have authority over the GSM when loading a sublevel, make sure we have authority
+	// over the actors in the sublevel.
+	if (GlobalStateManager != nullptr)
+	{
+		if (GlobalStateManager->HasAuthority())
+		{
+			for (auto Actor : LoadedLevel->Actors)
+			{
+				Actor->Role = ROLE_Authority;
+				Actor->RemoteRole = ROLE_SimulatedProxy;
+			}
+		}
+	}
+
+	// if we are the client, it is possible to have unresolved references
+	// to stably named actors in a newly loaded sublevel. Resolve these here.
+	if (bConnectAsClient && LoadedLevel->bIsVisible)
```

Comment:
When is bIsVisible false in this callback?

Reply:
I think it's more of a sanity check. I probably can remove it.